File: shared/enums.ts
import {z} from 'zod';

export const StorageDomain = z.enum(["config","extractions","aggregations","reductions","computations","mean_distance_matrix","autoclusters","metrics","trajectories","relative_trajectories"]);
// eslint-disable-next-line no-redeclare
export type StorageDomain = z.infer<typeof StorageDomain>;

export const ComputationStrategy = z.enum(["UMAP","PCA","EMBEDDINGS"]);
// eslint-disable-next-line no-redeclare
export type ComputationStrategy = z.infer<typeof ComputationStrategy>;

export const ExtractorImpl = z.enum(["BIRDNET","PERCH","SURF_PERCH","VGGISH","YAMNET","MUSIC_CLASS","SPECTRUM","SPECTROGRAM","MPS","MFCC","NDSI","BI","ADI","HF","HT","MED","ACI","LEQ","LEQ_PERCENTILE","LEQ_DIFF"]);
// eslint-disable-next-line no-redeclare
export type ExtractorImpl = z.infer<typeof ExtractorImpl>;

export const AutoclusterImpl = z.enum(["HDBSCAN_EOM","HDBSCAN_LEAF"]);
// eslint-disable-next-line no-redeclare
export type AutoclusterImpl = z.infer<typeof AutoclusterImpl>;

export const ReducerImpl = z.enum(["UMAP","PCA"]);
// eslint-disable-next-line no-redeclare
export type ReducerImpl = z.infer<typeof ReducerImpl>;

export const MetricImpl = z.enum(["MEAN_STD","MEAN_SPREADING","SILHOUETTE","CONTINGENCY","OVERLAP"]);
// eslint-disable-next-line no-redeclare
export type MetricImpl = z.infer<typeof MetricImpl>;

export const MetricType = z.enum(["ONE_D","TWO_D","TWO_D_PAIRING"]);
// eslint-disable-next-line no-redeclare
export type MetricType = z.infer<typeof MetricType>;

export const StftWindowType = z.enum(["HANN","HAMMING","BLACKMANHARRIS"]);
// eslint-disable-next-line no-redeclare
export type StftWindowType = z.infer<typeof StftWindowType>;

export const FrequencyScale = z.enum(["LIN","LOG","MEL"]);
// eslint-disable-next-line no-redeclare
export type FrequencyScale = z.infer<typeof FrequencyScale>;

export const AdiImpl = z.enum(["SHANNON","SIMPSON","INVSIMPSON"]);
// eslint-disable-next-line no-redeclare
export type AdiImpl = z.infer<typeof AdiImpl>;

File: processing/processing/enums.py
from enum import Enum


class StorageDomain(Enum):
    config = "config"
    extractions = "extractions"
    aggregations = "aggregations"
    reductions = "reductions"
    computations = "computations"
    mean_distance_matrix = "mean_distance_matrix"
    autoclusters = "autoclusters"
    metrics = "metrics"
    trajectories = "trajectories"
    relative_trajectories = "relative_trajectories"


class ExtractionStoragePath(Enum):
    embeddings = "embeddings"
    starts = "starts"
    ends = "ends"


class AggregationStoragePath(Enum):
    embeddings = "embeddings"
    timestamps = "timestamps"
    file_indices = "file_indices"
    file_relative_starts = "file_relative_starts"
    extractor_indices = "extractor_indices"


class TrajectoryStoragePath(Enum):
    path = "path"
    timestamps = "timestamps"


class RelativeTrajectoryStoragePath(Enum):
    distances = "distances"
    timestamps = "timestamps"
    deciles = "deciles"


class ComputationStrategy(Enum):
    UMAP = "UMAP"
    PCA = "PCA"
    EMBEDDINGS = "EMBEDDINGS"


class ExtractorImpl(Enum):
    BIRDNET = "BIRDNET"
    PERCH = "PERCH"
    SURF_PERCH = "SURF_PERCH"
    VGGISH = "VGGISH"
    YAMNET = "YAMNET"
    MUSIC_CLASS = "MUSIC_CLASS"
    SPECTRUM = "SPECTRUM"
    SPECTROGRAM = "SPECTROGRAM"
    MPS = "MPS"
    MFCC = "MFCC"
    NDSI = "NDSI"
    BI = "BI"
    ADI = "ADI"
    HF = "HF"
    HT = "HT"
    MED = "MED"
    ACI = "ACI"
    LEQ = "LEQ"
    LEQ_PERCENTILE = "LEQ_PERCENTILE"
    LEQ_DIFF = "LEQ_DIFF"


class AutoclusterImpl(Enum):
    HDBSCAN_EOM = "HDBSCAN_EOM"
    HDBSCAN_LEAF = "HDBSCAN_LEAF"


class ReducerImpl(Enum):
    UMAP = "UMAP"
    PCA = "PCA"


class MetricImpl(Enum):
    MEAN_STD = "MEAN_STD"
    MEAN_SPREADING = "MEAN_SPREADING"
    SILHOUETTE = "SILHOUETTE"
    CONTINGENCY = "CONTINGENCY"
    OVERLAP = "OVERLAP"


class MetricType(Enum):
    ONE_D = "ONE_D"
    TWO_D = "TWO_D"
    TWO_D_PAIRING = "TWO_D_PAIRING"


class StftWindowType(Enum):
    HANN = "HANN"
    HAMMING = "HAMMING"
    BLACKMANHARRIS = "BLACKMANHARRIS"


class FrequencyScale(Enum):
    LIN = "LIN"
    LOG = "LOG"
    MEL = "MEL"


class AdiImpl(Enum):
    SHANNON = "SHANNON"
    SIMPSON = "SIMPSON"
    INVSIMPSON = "INVSIMPSON"

